script.js

```diff
@@ -1,4 +1,523 @@
+  return Math.floor(Math.random() * max);
+};
+
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+var displayCards = function (player) {
+  var message = "";
+  if (player == numOfPlayers) {
+    player = "dealer";
+  }
+  for (var i = 0; i < master[player].decks.length; i += 1) {
+    message += `<br>${master[player].decks[i].name} of ${master[player].decks[i].emoji}`;
+  }
+  return message;
+};
+
+var displayOneDealerCard = function () {
+  var message = `<br><br>❗️One of the Dealer's Cards is ${master["dealer"].decks[0].name} of ${master["dealer"].decks[0].emoji}❗️`;
+  return message;
+};
+
+var displaySplitCards = function (player) {
+  var message = "";
+  master[player].blackjackValues = [];
+  for (var i = 0; i < master[player].decks.length; i += 1) {
+    var aceCounter = 0;
+    var total = 0;
+    var handNum = i + 1;
+    message += `<br><b>Hand ${handNum}</b>:`;
+    for (var j = 0; j < master[player].decks[i].length; j += 1) {
+      message += `<br>${master[player].decks[i][j].name} of ${master[player].decks[i][j].emoji}`;
+      if (master[player].decks[i][j].name == "ace" && aceCounter == 0) {
+        total += 11;
+        aceCounter += 1;
+      } else {
+        total += master[player].decks[i][j].blackjackValue;
+      }
+    }
+    if (aceCounter != 0 && total > 21) {
+      total -= 10;
+    }
+    if (total > 21) {
+      master[player].blackjackValues.push(0);
+    } else {
+      master[player].blackjackValues.push(total);
+    }
+
+    message += `<br>Hand's card value is <b>${total}</b><br>`;
+  }
+  return message;
+};
+
+var currentValue = function (player) {
+  var total = 0;
+  var aceCounter = 0;
+  if (player == numOfPlayers) {
+    player = "dealer";
+  }
+  for (var i = 0; i < master[player].decks.length; i += 1) {
+    if (master[player].decks[i].name == "ace" && aceCounter == 0) {
+      total += 11;
+      aceCounter += 1;
+    } else {
+      total += master[player].decks[i].blackjackValue;
+    }
+  }
+  if (aceCounter != 0 && total > 21) {
+    total -= 10;
+  }
+  return total;
+};
+
+var sumOfAllPoints = function (obj) {
+  var total = 0;
+  for (var i = 1; i < numOfPlayers; i += 1) {
+    total += obj[i].playerPoints;
+  }
+  return total;
+};
+
+var createPlayers = function (obj, numOfPlayers) {
+  for (var i = 0; i < numOfPlayers; i += 1) {
+    if (i + 1 == numOfPlayers) {
+      obj["dealer"] = { decks: [], blackjackValues: [] };
+    } else {
+      obj[i + 1] = {
+        decks: [],
+        blackjackValues: [],
+        playerPoints: 100,
+        playerBets: 0,
+        split: false,
+      };
+    }
+  }
+  return obj;
```

Comment:
is your index 0 of the array empty?
I have a feeling that this would lead to one less player than intended